helics/capi.py

```diff
@@ -9364,7 +9367,8 @@ def helicsQueryBufferFill(buffer: HelicsQueryBuffer, string: str):
     """
     f = loadSym("helicsQueryBufferFill")
     err = helicsErrorInitialize()
-    f(buffer.handle, string, len(string), err)
+    str_ptr = ffi.new('char[]', string.encode('utf-8'))
```

Comment:
Is HELICS expecting a utf-8 encoded string?

Reply:
This is what Mark Eberlein put in and it works so I guess, "yes"? It might make more sense to choose `ascii`. Let me know if you want me to change something.

Reply:
It looks like the test code in HELICS is just shoving a regular `char *` into the query buffer, which seems to imply that any encoding is possible? I think this function should have an argument that specifies the encoding to use (or no encoding/raw bytes). @phlptp

Reply:
@phlptp; any input on what you think we should do here? I don't have a philosophical problem with supporting data structures (raw bytes) as query responses if there's a need to do so **BUT** I suspect restricting this to ASCII or UTF-8 will make it easier to maintain the code and easier for users to understand what they are doing. Off the top of my head, I can't think of a use case where it would be important to support a raw bytes data structure where something like JSON wouldn't also be sufficient.

Reply:
I think I would be fine restricting it in python.   In the lower levels there are a few use cases for queries/commands getting raw binary data that I want to leave open, but that would be really complicated to use in python so don't really think we would need to support it, and if someone wanted and had the ability they could directly access the library calls.